### PR TITLE
Initial setup

### PR DESCRIPTION
**Pull request by AlexAxthelm**
Setup core workflow mechanics

* Accept JSON params
* Run workflow pacta
* run workflow pacta report

NOTE: Does not run R CMD CHECK on `windows-latest`

### CODE REVIEW

**Comment by AlexAxthelm** [comment]
Notes for reviewers: This is an MVP checkpoint. The next big step will be harmonizing this, `workflow.pacta`, and `workflow.pacta.report` to use the same schemas (or more accurately, sub/supersets of each other), but that's best done by opening up a triplet PR across the 3 repos, and modfiying them simultaneously.

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    needs: [docker]
+    secrets: inherit
+    strategy:
+      fail-fast: false
+      matrix:
+        config-name:
+          # - default_2022Q4
+          # - default_2023Q4
+          - full_params_2022Q4
+          - full_params_2023Q4
```
Comment (jdhoffa) [inline .github/workflows/docker.yml:29]: NB: What's the difference between `default_*` and `full_params_*`?

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        target: /mnt/benchmarks_dir/
+        read_only: true
+      - type: bind
+        source: ${analysis_output_dir:-./analysis_output_dir/}
+        target: /mnt/analysis_output_dir
+        read_only: false
+      - type: bind
+        source: ${real_estate_dir:-./real_estate_dir/}
+        target: /mnt/real_estate_dir
+        read_only: true
```
Comment (jdhoffa) [inline docker-compose.yml:40]: Pretty sure this is safe to remove? Real estate has been wiped from PACTA (e.g. https://github.com/RMI-PACTA/workflow.transition.monitor/pull/308)

We're probably going to need to manually review all the latest commits in `workflow.transition.monitor` and `workflow.pacta`/`workflow.pacta.report` to make sure the have all the same recent updates...

**Review by jdhoffa — APPROVED** [review]
NB comments, but looks great

File: docker-compose.yml (hunk 2 above)
Comment (jdhoffa) [reply docker-compose.yml:40]: @AlexAxthelm do you mind opening a new issue(s) to track that?

File: .github/workflows/docker.yml (hunk 1 above)
Comment (AlexAxthelm) [reply .github/workflows/docker.yml:29]: The `default_*`s use the `inherit` key (which should be the main mechanism), but that isn't setup in this repo yet (so they're commented out). the `full_` ones send what should be the same parameters after the param inheritance process runs, but does it all as raw JSON input.

The corresponding directory in [`workflow.pacta`](https://github.com/RMI-PACTA/workflow.pacta/tree/main/tests/config) give a pretty good idea.

File: docker-compose.yml (hunk 2 above)
Comment (AlexAxthelm) [reply docker-compose.yml:40]: #6

Comment (jdhoffa) [reply docker-compose.yml:40]: Thanks!

**Pull request merged**